diff --git a/lib/module/views/useHeaderConfigProps.js b/lib/module/views/useHeaderConfigProps.js
index a56cb12fc83995973fad7d9e851d6fdbaf999a17..c3b83008dcd5a7bc7c3011bbf24363fdddb3f2ae 100644
--- a/lib/module/views/useHeaderConfigProps.js
+++ b/lib/module/views/useHeaderConfigProps.js
@@ -19,6 +19,12 @@ const processBarButtonItems = (items, colors, fonts) => {
       }
       return item;
     }
+    if (item.type === 'mailSearchToolbar') {
+      return {
+        ...item,
+        index
+      };
+    }
     if (item.type === 'button' || item.type === 'menu') {
       if (item.type === 'menu' && item.menu == null) {
         throw new Error(`Menu item must have a 'menu' property defined: ${JSON.stringify(item)}`);
@@ -79,7 +85,7 @@ const processBarButtonItems = (items, colors, fonts) => {
       }
       return processedItem;
     }
-    throw new Error(`Invalid item type: ${JSON.stringify(item)}. Valid types are 'button', 'menu', 'custom' and 'spacing'.`);
+    throw new Error(`Invalid item type: ${JSON.stringify(item)}. Valid types are 'button', 'menu', 'custom', 'spacing' and 'mailSearchToolbar'.`);
   }).filter(item => item != null);
 };
 const transformIcon = icon => {
@@ -158,9 +164,13 @@ export function useHeaderConfigProps({
   headerBack,
   route,
   title,
+  unstable_headerCenterItems: headerCenterItems,
   unstable_headerInsets: headerInsets,
   unstable_headerLeftItems: headerLeftItems,
-  unstable_headerRightItems: headerRightItems
+  unstable_headerRightItems: headerRightItems,
+  unstable_headerSubtitle: headerSubtitle,
+  unstable_headerToolbarItems: headerToolbarItems,
+  unstable_navigationItemStyle: navigationItemStyle
 }) {
   const {
     direction
@@ -256,6 +266,10 @@ export function useHeaderConfigProps({
     tintColor,
     canGoBack
   });
+  const centerItems = headerCenterItems?.({
+    tintColor,
+    canGoBack
+  });
   let rightItems = headerRightItems?.({
     tintColor,
     canGoBack
@@ -265,6 +279,10 @@ export function useHeaderConfigProps({
     // So we need to reverse them here to match the order
     rightItems = [...rightItems].reverse();
   }
+  const toolbarItems = headerToolbarItems?.({
+    tintColor,
+    canGoBack
+  });
   const children = /*#__PURE__*/_jsxs(_Fragment, {
     children: [Platform.OS === 'ios' ? /*#__PURE__*/_jsxs(_Fragment, {
       children: [leftItems ? leftItems.map((item, index) => {
@@ -279,7 +297,15 @@ export function useHeaderConfigProps({
         return null;
       }) : headerLeftElement != null ? /*#__PURE__*/_jsx(ScreenStackHeaderLeftView, {
         children: headerLeftElement
-      }) : null, headerTitleElement != null ? /*#__PURE__*/_jsx(ScreenStackHeaderCenterView, {
+      }) : null, centerItems ? centerItems.map((item, index) => {
+        if (item.type === 'custom') {
+          return /*#__PURE__*/_jsx(ScreenStackHeaderCenterView, {
+            hidesSharedBackground: item.hidesSharedBackground,
+            children: item.element
+          }, index);
+        }
+        return null;
+      }) : headerTitleElement != null ? /*#__PURE__*/_jsx(ScreenStackHeaderCenterView, {
         children: headerTitleElement
       }) : null]
     }) : /*#__PURE__*/_jsxs(_Fragment, {
@@ -357,6 +383,7 @@ export function useHeaderConfigProps({
     largeTitleFontWeight,
     largeTitleHideShadow: headerLargeTitleShadowVisible === false,
     title: titleText,
+    subtitle: headerSubtitle,
     titleColor,
     titleFontFamily,
     titleFontSize,
@@ -368,6 +395,9 @@ export function useHeaderConfigProps({
     disableBottomInsetApplication: headerInsets?.bottom !== undefined ? !headerInsets.bottom : undefined,
     translucent: translucent === true,
     children,
+    navigationItemStyle,
+    headerToolbarItems: processBarButtonItems(toolbarItems, colors, fonts),
+    headerCenterBarButtonItems: processBarButtonItems(centerItems, colors, fonts),
     headerLeftBarButtonItems: processBarButtonItems(leftItems, colors, fonts),
     headerRightBarButtonItems: processBarButtonItems(rightItems, colors, fonts),
     experimental_userInterfaceStyle: dark ? 'dark' : 'light'
